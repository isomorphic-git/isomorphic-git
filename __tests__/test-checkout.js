/* globals describe test expect */
// import fs from 'fs'
const { makeFixture } = require('./__helpers__/FixtureFS.js')
// import { createTempDir, copyFixtureIntoTempDir } from 'jest-fixtures'
import pify from 'pify'
import { checkout, listFiles } from '..'

/** @test {checkout} */
describe('checkout', () => {
  test('checkout', async () => {
    let { fs, dir, gitdir } = await makeFixture('test-checkout')
    // let dir = await createTempDir()
    // let gitdir = await copyFixtureIntoTempDir(__dirname, 'test-checkout.git')
    // let repo = { fs, dir, gitdir }
    await checkout({ fs, dir, gitdir, ref: 'test-branch' })
    let files = await pify(fs.readdir)(dir)
    expect(files.sort()).toMatchSnapshot()
    let index = await listFiles({ fs, dir, gitdir })
    expect(index).toMatchSnapshot()
  })
})
