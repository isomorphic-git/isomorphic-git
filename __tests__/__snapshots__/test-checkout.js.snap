// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkout checkout 1`] = `
Array [
  ".babelrc",
  ".editorconfig",
  ".flowconfig",
  ".gitignore",
  "LICENSE.md",
  "README.md",
  "package.json",
  "shrinkwrap.yaml",
  "src",
  "test",
]
`;

exports[`checkout checkout 2`] = `
Array [
  ".babelrc",
  ".editorconfig",
  ".flowconfig",
  ".gitignore",
  "LICENSE.md",
  "README.md",
  "package.json",
  "shrinkwrap.yaml",
  "src/commands/checkout.js",
  "src/commands/config.js",
  "src/commands/fetch.js",
  "src/commands/init.js",
  "src/index.js",
  "src/models/GitBlob.js",
  "src/models/GitCommit.js",
  "src/models/GitConfig.js",
  "src/models/GitTree.js",
  "src/utils/combinePayloadAndSignature.js",
  "src/utils/commitSha.js",
  "src/utils/exists.js",
  "src/utils/mkdirs.js",
  "src/utils/read.js",
  "src/utils/resolveRef.js",
  "src/utils/unwrapObject.js",
  "src/utils/wrapCommit.js",
  "src/utils/write.js",
  "test/resolveRef.js",
  "test/smoke.js",
  "test/snapshots/resolveRef.js.md",
  "test/snapshots/resolveRef.js.snap",
]
`;

exports[`checkout checkout by SHA 1`] = `
Array [
  ".babelrc",
  ".editorconfig",
  ".flowconfig",
  ".gitignore",
  "LICENSE.md",
  "README.md",
  "package.json",
  "shrinkwrap.yaml",
  "src",
  "test",
]
`;

exports[`checkout checkout by SHA 2`] = `
Array [
  ".babelrc",
  ".editorconfig",
  ".flowconfig",
  ".gitignore",
  "LICENSE.md",
  "README.md",
  "package.json",
  "shrinkwrap.yaml",
  "src/commands/checkout.js",
  "src/commands/config.js",
  "src/commands/fetch.js",
  "src/commands/init.js",
  "src/index.js",
  "src/models/GitBlob.js",
  "src/models/GitCommit.js",
  "src/models/GitConfig.js",
  "src/models/GitTree.js",
  "src/utils/combinePayloadAndSignature.js",
  "src/utils/commitSha.js",
  "src/utils/exists.js",
  "src/utils/mkdirs.js",
  "src/utils/read.js",
  "src/utils/resolveRef.js",
  "src/utils/unwrapObject.js",
  "src/utils/wrapCommit.js",
  "src/utils/write.js",
  "test/resolveRef.js",
  "test/smoke.js",
  "test/snapshots/resolveRef.js.md",
  "test/snapshots/resolveRef.js.snap",
]
`;

exports[`checkout checkout by tag 1`] = `
Array [
  ".babelrc",
  ".editorconfig",
  ".flowconfig",
  ".gitignore",
  "LICENSE.md",
  "README.md",
  "package.json",
  "shrinkwrap.yaml",
  "src",
  "test",
]
`;

exports[`checkout checkout by tag 2`] = `
Array [
  ".babelrc",
  ".editorconfig",
  ".flowconfig",
  ".gitignore",
  "LICENSE.md",
  "README.md",
  "package.json",
  "shrinkwrap.yaml",
  "src/commands/checkout.js",
  "src/commands/config.js",
  "src/commands/fetch.js",
  "src/commands/init.js",
  "src/index.js",
  "src/models/GitBlob.js",
  "src/models/GitCommit.js",
  "src/models/GitConfig.js",
  "src/models/GitTree.js",
  "src/utils/combinePayloadAndSignature.js",
  "src/utils/commitSha.js",
  "src/utils/exists.js",
  "src/utils/mkdirs.js",
  "src/utils/read.js",
  "src/utils/resolveRef.js",
  "src/utils/unwrapObject.js",
  "src/utils/wrapCommit.js",
  "src/utils/write.js",
  "test/resolveRef.js",
  "test/smoke.js",
  "test/snapshots/resolveRef.js.md",
  "test/snapshots/resolveRef.js.snap",
]
`;

exports[`checkout checkout directories using filepaths 1`] = `
Array [
  "src",
  "test",
]
`;

exports[`checkout checkout directories using filepaths 2`] = `
Array [
  "src/models/GitBlob.js",
  "src/models/GitCommit.js",
  "src/models/GitConfig.js",
  "src/models/GitTree.js",
  "test/resolveRef.js",
  "test/smoke.js",
  "test/snapshots/resolveRef.js.md",
  "test/snapshots/resolveRef.js.snap",
]
`;

exports[`checkout checkout files using filepaths 1`] = `
Array [
  "src",
]
`;

exports[`checkout checkout files using filepaths 2`] = `
Array [
  "src/models/GitBlob.js",
  "src/utils/write.js",
]
`;

exports[`checkout checkout files using filepaths and deep pattern 1`] = `
Array [
  "test",
]
`;

exports[`checkout checkout files using filepaths and deep pattern 2`] = `
Array [
  "test/snapshots/resolveRef.js.md",
  "test/snapshots/resolveRef.js.snap",
]
`;

exports[`checkout checkout files using filepaths and pattern 1`] = `
Array [
  "src",
  "test",
]
`;

exports[`checkout checkout files using filepaths and pattern 2`] = `
Array [
  "src/utils/read.js",
  "src/utils/resolveRef.js",
  "test/resolveRef.js",
]
`;

exports[`checkout checkout unfetched branch 1`] = `
Object {
  "caller": "git.checkout",
  "code": "CommitNotFetchedError",
  "data": Object {
    "oid": "033417ae18b174f078f2f44232cb7a374f4c60ce",
    "ref": "missing-branch",
  },
  "message": "Failed to checkout \\"missing-branch\\" because commit 033417ae18b174f078f2f44232cb7a374f4c60ce is not available locally. Do a git fetch to make the branch available locally.",
}
`;
